upip/upip.py

```diff
@@ -243,6 +249,113 @@ def cleanup():
         except OSError:
             print("Warning: Cannot delete " + fname)
 
+def parse_version(string):
```

Comment:
Should probably have unit tests around this method. I may be able to help there...
Also, consider changing the variable `string` since it's very generic. Perhaps `version_string`?

I'll try and test this (on the unix and ESP32 ports) soon. I've also only had a superficial glance over the parsing logic; will try to understand it better.